src/verilog/verilog_synthesis.cpp

```diff
@@ -53,60 +51,72 @@ void verilog_synthesist::synth_expr(
     {
       exprt c=from_integer(v_it->second, integer_typet());
       c.add_source_location()=expr.source_location();
-      expr=c;
+      return c;
     }
     else if(symbol.is_macro)
     {
       // substitute
       assert(symbol.value.is_not_nil());
-      expr=symbol.value;
-      
+
       // recursive call
-      synth_expr(expr, symbol_state);
+      return synth_expr(symbol.value, symbol_state);
     }
     else
     {
       switch(symbol_state)
       {
       case symbol_statet::SYMBOL:
-        break;
+        return expr; // leave as is
 
       case symbol_statet::CURRENT:
-        expr=current_value(symbol);
-        break;
+        return current_value(symbol);
 
       case symbol_statet::FINAL:
-        expr=final_value(symbol);
-        break;
+        return final_value(symbol);
 
       case symbol_statet::NONE:
         error().source_location=expr.source_location();
         error() << "symbols not allowed here" << eom;
         throw 0;
       }
+
+      UNREACHABLE;
```

Comment:
Should be unnecessary as we compile with warnings about missing cases.

Reply:
Neither clang nor gcc worked that out.

Reply:
Correction: clang does work that out, gcc doesn't.